writer.go

```diff
@@ -358,11 +545,11 @@ func (w *writer) decodeField(src []byte) ([]byte, bool) {
 			}
 			w.Write("\"")
 			delimited = nil
-			goto justBytes
+			goto decodeBytes
 		}
 
 		// Who knows what it is? Bytes or something.
-	justBytes:
+	decodeBytes:
```

Comment:
I wonder if tail calls to well named helpers would be easier to follow than gotos here